client/src/main/java/tds/assessment/Assessment.java

```diff
@@ -51,7 +61,7 @@ public String getAssessmentId() {
      * @return {@code true} if the assessment is segmented
      */
     public boolean isSegmented() {
-        return segmented;
+        return segments.size() > 1;
```

Comment:
Suggest doing `return segments.isEmpty()` and add a check in the constructor that the List has to either be empty or contain more than 1 record.